src/nlp/utils/py_utils.py

```diff
@@ -268,14 +270,29 @@ class Pickler(dill.Pickler):
 
 def dump(obj, file):
     """pickle an object to a file"""
-    Pickler(file).dump(obj)
+    Pickler(file, recurse=True).dump(obj)
     return
 
 
+@contextlib.contextmanager
+def _no_cache_fields(obj):
+    if _transformers_available:
+        import transformers as tr
+
+        if isinstance(obj, (tr.CTRLTokenizer, tr.GPT2Tokenizer, tr.OpenAIGPTTokenizer, tr.XLMTokenizer)):
```

Comment:
This list might be a bit cumbersome to maintain in the future.
Should we just do a check that the `cache` attribute exists and is a dict maybe?

Reply:
I changed it so that it checks if `cache` exists and is a dict